src/main/java/com/odysseusinc/arachne/portal/security/AuthenticationTokenFilter.java

```diff
@@ -74,6 +76,10 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
             }
             if (authToken != null) {
                 String username = this.tokenUtils.getUsernameFromToken(authToken);
+                String requested = httpRequest.getHeader(USER_REQUEST_HEADER);
+                if (requested != null && username != null && !Objects.equals(username, requested)){
+                    throw new BadCredentialsException("forced logout");
```

Comment:
Won't this spam with a stacktrace?

Reply:
There is catch below that prints stacktrace in debug level only:
```java
} catch (AuthenticationException ex) {
            logger.debug(ex.getMessage(), ex);
```